src/main/java/com/owncloud/android/lib/common/network/WebdavEntry.java

```diff
@@ -319,12 +327,80 @@ public WebdavEntry(MultiStatusResponse ms, String splitElement) {
             // NC sharees property <nc-sharees>
             prop = propSet.get(EXTENDED_PROPERTY_SHAREES, ncNamespace);
             if (prop != null && prop.getValue() != null) {
-                Collections.addAll(sharees, prop.getValue().toString().split(", "));
+                if (prop.getValue() instanceof ArrayList) {
+                    ArrayList list = (ArrayList) prop.getValue();
+
+                    List<ShareeUser> tempList = new ArrayList<>();
+                    
+                    for (int i = 0; i < list.size(); i++) {
+                        Element element = (Element) list.get(i);
+
+                        ShareeUser user = createShareeUser(element);
+
+                        if (user != null) {
+                            tempList.add(user);
+                        }
+                    }
+
+                    sharees = tempList.toArray(new ShareeUser[0]);
+
+                } else {
+                    // single item or empty
+                    Element element = (Element) prop.getValue();
+
+                    ShareeUser user = createShareeUser(element);
+
+                    if (user != null) {
+                        sharees = new ShareeUser[]{user};
+                    }
+                }
             }
         } else {
             Log_OC.e("WebdavEntry", "General fuckup, no status for webdav response");
         }
     }
+
+    private @Nullable
+    ShareeUser createShareeUser(Element element) {
```

Comment:
should be a single line

`private @Nullable ShareeUser createShareeUser(Element element) {`